### PR TITLE
No need to create a new list just so you can use List<T>.IndexOf(..) …

### PR DESCRIPTION
**Pull request by DaEgi01**
…if you can use Array.FindIndex(..) instead.

### CODE REVIEW

**Review by kvakvs — APPROVED** [review]
nice

**Review by krzychu124 — APPROVED** [review]

File: TLM/CSUtil.Commons/Log.cs [hunk 1]
```diff
@@ -62,8 +61,8 @@ private enum LogLevel {
                     File.Delete(LogFilePath); // delete old file to avoid confusion.
                 }
 
-                var args = Environment.GetCommandLineArgs().ToList();
-                int index = args.IndexOf("-logFile");
+                var args = Environment.GetCommandLineArgs();
+                int index = Array.FindIndex(args, x => x == "-logFile");
```
Comment (kianzarrin) [inline TLM/CSUtil.Commons/Log.cs:65]: this is still allocating a delegate
you can do `(args as IList<string>).IndexOf("-logFile");`

but either way I don't care!

Comment (kianzarrin) [reply TLM/CSUtil.Commons/Log.cs:65]: I wish C# did not require me to cast array as IList to use IList extension methods.
I don't have to cast it to IEnumerable to use IEnumerable extension methods for that.

Comment (kianzarrin) [reply TLM/CSUtil.Commons/Log.cs:65]: you cannot add/remove stuff from array so maybe that is why.
but still array can be casted to IList.
and trying to add remove stuff throws exception:
```
(x as IList<string>).Add("zarrin")
System.NotSupportedException: Collection was of a fixed size.
```

**Pull request merged**